### PR TITLE
feat: readonly fields

### LINKED ISSUE #376: feat: add support for readonly input fields
## Description

It is sometimes beneficial to display data in an input field to make represent that there is some data in the form which is not editable.

### PR DESCRIPTION
**Pull request by SalihuDickson**
## Description

Fixes #376

## Checklist

- [ ] My code follows the [contributing guidelines][contributing] of this project.
- [ ] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

## Summary by Sourcery

Add a 'readonly' option to form fields to allow them to be disabled, and enhance tooltip handling by trimming whitespace.

New Features:
- Introduce a 'readonly' field option to form components, allowing fields to be displayed as disabled when specified.

Enhancements:
- Improve tooltip handling by trimming whitespace to ensure tooltips are only displayed when content is present.

### CODE REVIEW

File: packages/ecc-utils-design/src/components/form/form.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface Field {
     accept?: string;
     returnIfEmpty?: boolean;
     tooltip?: string;
+    readonly?: string;
```
Comment (anuragxxd) [inline packages/ecc-utils-design/src/components/form/form.ts:44]: This can be a `boolean`!

**Review by anuragxxd — CHANGES_REQUESTED** [review]
lgtm @SalihuDickson, just a small comment. 

Also can you maybe add the [docs](https://github.com/elixir-cloud-aai/cloud-components/pull/380#issuecomment-2453074985) & the changeset for the same too? This will put this is the changelog.

**Comment by SalihuDickson** [comment]
sure @anuragxxd, i'll add the feature to the docs.

File: packages/ecc-utils-design/src/components/form/form.ts (hunk 1 above)
Comment (SalihuDickson) [reply packages/ecc-utils-design/src/components/form/form.ts]: 😅 yh, didn't realize I made it a string

**Comment by SalihuDickson** [comment]
@sivangbagri, when this get's merged please add create a PR to add a single type field back the dataset entity, and it should be a readonly field.

**Comment by SalihuDickson** [comment]
@anuragxxd, issues have been resolved, please review.

**Review by anuragxxd — APPROVED** [review]

**Pull request merged**